lib/base.ts

```diff
@@ -36,6 +36,13 @@ export class TranspilerBase {
     }
   }
 
+  assert(c: ts.Node, condition: boolean, reason: string): void {
+    if (!condition) {
+      this.reportError(c, reason);
+      throw new Error(reason);
```

Comment:
Assertions should stop the execution flow. In the case of Promises (where it's first used), assertion means that we already know the code generated will not work.